onnxruntime/core/providers/xnnpack/nn/gemm.h

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/framework/op_kernel.h"
+#include "core/framework/allocator.h"
+#include "core/providers/cpu/math/gemm_base.h"
+#include "core/providers/xnnpack/detail/utils.h"
+#include "xnnpack.h"
+#include "core/common/common.h"
+#include "core/util/math.h"
+
+namespace onnxruntime {
+class GraphViewer;
+class Node;
+namespace xnnpack {
+
+class Gemm : protected GemmBase, public OpKernel {
+ public:
+  Gemm(const OpKernelInfo& info);
+
+  Status Compute(OpKernelContext* /*context*/) const override;
+
+  static bool IsGemmOnnxNodeSupported(const NodeUnit& node_unit, const GraphViewer& graph);
+
+  Status PrePack(const Tensor& tensor, int input_idx, AllocatorPtr alloc,
+                 /*out*/ bool& is_packed,
+                 /*out*/ PrePackedWeights* prepacked_weights) override;
+
+  Status CreateXnnpackOpp(
+      size_t input_channels,
+      size_t output_channels,
+      size_t input_stride,
+      size_t output_stride,
+      const float* kernel,
+      const float* bias,
+      float output_min,
+      float output_max,
+      uint32_t flags);
+
+ private:
+
+  TensorShape b_shape_;
+  BufferUniquePtr packed_b_=nullptr;
```

Comment:
nit: please format with clang-format. Also a unique_ptr doesn't need explicit initialization.

Reply:
Updated in following commit - I removed the initialization